### DISCUSSION TITLE
Custom @Filter not recognized on start

### DISCUSSION THREAD

**Discussion by AdiWehrli** [tchiotludo/akhq#1234] [Q&A]
Hello everybody

I created a custom HTTP filter with the `@Filter` annotation following the [Micronaut documention](https://docs.micronaut.io/latest/guide/index.html#filters). I added the corresponding JAR file to the classpath in AKHQ but on start AKHQ/Micronaut does not recognize it. In my unit tests Micronaut creates the filter (bean) successfully as the log reports.

Have I forgotten something? It is said that it should be sufficient to annotate the filter with `@Filter` and put it into the classpath. Or would it be necessary to include the filter class into the `akhq.jar` file?

Thanks very much in advance.

Best regards,
Adi

**Comment by tchiotludo** [comment]
Hey, 
to be honest, I think that should work if the classpath is respected but I'm not really sure about that. 
What is the goal of this `@Filter`?

**Reply by AdiWehrli** [reply]
The goal is to add custom headers to the HTTP response.

**Reply by tchiotludo** [reply]
You can make a PR with the ability to configure it on `application.yml` it will be merged !

**Reply by AdiWehrli** [reply]
Done: https://github.com/tchiotludo/akhq/pull/1235
